prow/plugins/label/label.go

```diff
@@ -96,20 +102,24 @@ func getLabelsFromREMatches(matches [][]string) (labels []string) {
 
 // getLabelsFromGenericMatches returns label matches with extra labels if those
 // have been configured in the plugin config.
-func getLabelsFromGenericMatches(matches [][]string, additionalLabels []string) []string {
+func getLabelsFromGenericMatches(matches [][]string, additionalLabels []string, invalidLabels *[]string) []string {
 	if len(additionalLabels) == 0 {
 		return nil
 	}
 	var labels []string
+	labelFilter := sets.String{}
+	for _, l := range additionalLabels {
+		labelFilter.Insert(strings.ToLower(l))
+	}
 	for _, match := range matches {
 		parts := strings.Split(match[0], " ")
 		if ((parts[0] != "/label") && (parts[0] != "/remove-label")) || len(parts) != 2 {
 			continue
 		}
-		for _, l := range additionalLabels {
-			if l == parts[1] {
-				labels = append(labels, parts[1])
-			}
+		if labelFilter.Has(strings.ToLower(parts[1])) {
```

Comment:
We should add a return type to this function instead of passing in a pointer to mutate

Reply:
I agree and I am happy to do it.

Incorporating the proposed change will trigger further changes to this file. I want to minimize changes to my first issue.

What if after the change I follow up, in a 2nd PR and add the proposed change, so that this function returns a type?

Reply:
Added an [issue to track this](https://github.com/kubernetes/test-infra/issues/14209).